File: push/src/main/java/com/xiaomi/mipush/sdk/PushMessageHelper.java
package com.xiaomi.mipush.sdk;

import android.content.Context;
import android.content.Intent;
import android.text.TextUtils;
import com.xiaomi.push.service.PushConstants;
import com.xiaomi.xmpush.thrift.PushMetaInfo;
import com.xiaomi.xmpush.thrift.XmPushActionSendMessage;
import java.util.List;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/mipush/sdk/PushMessageHelper.class */
public class PushMessageHelper {
    public static final String ERROR_MESSAGE = "error_message";
    public static final String ERROR_TYPE = "error_type";
    public static final String ERROR_TYPE_NEED_PERMISSION = "error_lack_of_permission";
    public static final String KEY_COMMAND = "key_command";
    public static final String KEY_MESSAGE = "key_message";
    public static final int MESSAGE_COMMAND = 3;
    public static final int MESSAGE_ERROR = 5;
    public static final int MESSAGE_QUIT = 4;
    public static final int MESSAGE_RAW = 1;
    public static final int MESSAGE_SENDMESSAGE = 2;
    public static final String MESSAGE_TYPE = "message_type";
    public static final int PUSH_MODE_BROADCAST = 2;
    public static final int PUSH_MODE_CALLBACK = 1;
    private static int pushMode = 0;

    public static MiPushCommandMessage generateCommandMessage(String str, List<String> list, long j, String str2, String str3) {
        MiPushCommandMessage miPushCommandMessage = new MiPushCommandMessage();
        miPushCommandMessage.setCommand(str);
        miPushCommandMessage.setCommandArguments(list);
        miPushCommandMessage.setResultCode(j);
        miPushCommandMessage.setReason(str2);
        miPushCommandMessage.setCategory(str3);
        return miPushCommandMessage;
    }

    public static MiPushMessage generateMessage(XmPushActionSendMessage xmPushActionSendMessage, PushMetaInfo pushMetaInfo, boolean z) {
        MiPushMessage miPushMessage = new MiPushMessage();
        miPushMessage.setMessageId(xmPushActionSendMessage.getId());
        if (!TextUtils.isEmpty(xmPushActionSendMessage.getAliasName())) {
            miPushMessage.setMessageType(1);
            miPushMessage.setAlias(xmPushActionSendMessage.getAliasName());
        } else if (!TextUtils.isEmpty(xmPushActionSendMessage.getTopic())) {
            miPushMessage.setMessageType(2);
            miPushMessage.setTopic(xmPushActionSendMessage.getTopic());
        } else if (TextUtils.isEmpty(xmPushActionSendMessage.getUserAccount())) {
            miPushMessage.setMessageType(0);
        } else {
            miPushMessage.setMessageType(3);
            miPushMessage.setUserAccount(xmPushActionSendMessage.getUserAccount());
        }
        miPushMessage.setCategory(xmPushActionSendMessage.getCategory());
        if (xmPushActionSendMessage.getMessage() != null) {
            miPushMessage.setContent(xmPushActionSendMessage.getMessage().getPayload());
        }
        if (pushMetaInfo != null) {
            if (TextUtils.isEmpty(miPushMessage.getMessageId())) {
                miPushMessage.setMessageId(pushMetaInfo.getId());
            }
            if (TextUtils.isEmpty(miPushMessage.getTopic())) {
                miPushMessage.setTopic(pushMetaInfo.getTopic());
            }
            miPushMessage.setDescription(pushMetaInfo.getDescription());
            miPushMessage.setTitle(pushMetaInfo.getTitle());
            miPushMessage.setNotifyType(pushMetaInfo.getNotifyType());
            miPushMessage.setNotifyId(pushMetaInfo.getNotifyId());
            miPushMessage.setPassThrough(pushMetaInfo.getPassThrough());
            miPushMessage.setExtra(pushMetaInfo.getExtra());
        }
        miPushMessage.setNotified(z);
        return miPushMessage;
    }

    public static PushMetaInfo generateMessage(MiPushMessage miPushMessage) {
        PushMetaInfo pushMetaInfo = new PushMetaInfo();
        pushMetaInfo.setId(miPushMessage.getMessageId());
        pushMetaInfo.setTopic(miPushMessage.getTopic());
        pushMetaInfo.setDescription(miPushMessage.getDescription());
        pushMetaInfo.setTitle(miPushMessage.getTitle());
        pushMetaInfo.setNotifyId(miPushMessage.getNotifyId());
        pushMetaInfo.setNotifyType(miPushMessage.getNotifyType());
        pushMetaInfo.setPassThrough(miPushMessage.getPassThrough());
        pushMetaInfo.setExtra(miPushMessage.getExtra());
        return pushMetaInfo;
    }

    public static int getPushMode(Context context) {
        if (pushMode == 0) {
            if (isUseCallbackPushMode(context)) {
                setPushMode(1);
            } else {
                setPushMode(2);
            }
        }
        return pushMode;
    }

    private static boolean isIntentAvailable(Context context, Intent intent) {
        try {
            List listQueryBroadcastReceivers = context.getPackageManager().queryBroadcastReceivers(intent, 32);
            return listQueryBroadcastReceivers != null && !listQueryBroadcastReceivers.isEmpty();
        } catch (Exception e) {
            return true;
        }
    }

    public static boolean isUseCallbackPushMode(Context context) {
        Intent intent = new Intent(PushConstants.MIPUSH_ACTION_NEW_MESSAGE);
        intent.setClassName(context.getPackageName(), "com.xiaomi.mipush.sdk.PushServiceReceiver");
        return isIntentAvailable(context, intent);
    }

    public static void sendCommandMessageBroadcast(Context context, MiPushCommandMessage miPushCommandMessage) {
        Intent intent = new Intent(PushConstants.MIPUSH_ACTION_NEW_MESSAGE);
        intent.setPackage(context.getPackageName());
        intent.putExtra(MESSAGE_TYPE, 3);
        intent.putExtra(KEY_COMMAND, miPushCommandMessage);
        new PushServiceReceiver().onReceive(context, intent);
    }

    public static void sendQuitMessageBroadcast(Context context) {
        Intent intent = new Intent(PushConstants.MIPUSH_ACTION_NEW_MESSAGE);
        intent.setPackage(context.getPackageName());
        intent.putExtra(MESSAGE_TYPE, 4);
        new PushServiceReceiver().onReceive(context, intent);
    }

    private static void setPushMode(int i) {
        pushMode = i;
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/mipush/sdk/PushMessageHelper.kt
package com.xiaomi.mipush.sdk

import android.content.Context
import android.content.Intent
import android.text.TextUtils
import com.xiaomi.push.service.PushConstants
import com.xiaomi.xmpush.thrift.PushMetaInfo
import com.xiaomi.xmpush.thrift.XmPushActionSendMessage

/*
 * Current override reference: com.xiaomi.xmsf 0.3.17-20260410000745 (versionCode 1003003000),
 * base.apk sha256 f3d72b6f5e1427ceecd3147a051d58e4dc95bb528397d486658e01cad9f7e590,
 * JADX path: com.xiaomi.xmsf/current/base/sources/com/xiaomi/mipush/sdk/PushMessageHelper.java
 * No stock 7.4.67-C same-path source was found in the split source tree.
 */
object PushMessageHelper {
    const val ERROR_MESSAGE = "error_message"
    const val ERROR_TYPE = "error_type"
    const val ERROR_TYPE_NEED_PERMISSION = "error_lack_of_permission"
    const val KEY_COMMAND = "key_command"
    const val KEY_MESSAGE = "key_message"
    const val MESSAGE_COMMAND = 3
    const val MESSAGE_ERROR = 5
    const val MESSAGE_QUIT = 4
    const val MESSAGE_RAW = 1
    const val MESSAGE_SENDMESSAGE = 2
    const val MESSAGE_TYPE = "message_type"
    const val PUSH_MODE_BROADCAST = 2
    const val PUSH_MODE_CALLBACK = 1

    private var pushMode = 0
    private var sdkGateway: ISDKGateway? = null

    @JvmStatic
    fun setSDKGateway(gateway: ISDKGateway?) {
        sdkGateway = gateway
    }

    @JvmStatic
    fun generateCommandMessage(
        command: String?,
        arguments: List<String>?,
        resultCode: Long,
        reason: String?,
        category: String?
    ): MiPushCommandMessage {
        return MiPushCommandMessage().apply {
            this.command = command
            commandArguments = arguments
            this.resultCode = resultCode
            this.reason = reason
            this.category = category
        }
    }

    @JvmStatic
    fun generateMessage(sendMessage: XmPushActionSendMessage, metaInfo: PushMetaInfo?, notified: Boolean): MiPushMessage {
        val miPushMessage = MiPushMessage().apply {
            messageId = sendMessage.id
        }
        if (!TextUtils.isEmpty(sendMessage.aliasName)) {
            miPushMessage.messageType = 1
            miPushMessage.alias = sendMessage.aliasName
        } else if (!TextUtils.isEmpty(sendMessage.topic)) {
            miPushMessage.messageType = 2
            miPushMessage.topic = sendMessage.topic
        } else if (TextUtils.isEmpty(sendMessage.userAccount)) {
            miPushMessage.messageType = 0
        } else {
            miPushMessage.messageType = 3
            miPushMessage.userAccount = sendMessage.userAccount
        }
        miPushMessage.category = sendMessage.category
        if (sendMessage.message != null) {
            miPushMessage.content = sendMessage.message.payload
        }
        if (metaInfo != null) {
            if (TextUtils.isEmpty(miPushMessage.messageId)) {
                miPushMessage.messageId = metaInfo.id
            }
            if (TextUtils.isEmpty(miPushMessage.topic)) {
                miPushMessage.topic = metaInfo.topic
            }
            miPushMessage.description = metaInfo.description
            miPushMessage.title = metaInfo.title
            miPushMessage.notifyType = metaInfo.notifyType
            miPushMessage.notifyId = metaInfo.notifyId
            miPushMessage.passThrough = metaInfo.passThrough
            miPushMessage.extra = metaInfo.extra
        }
        miPushMessage.isNotified = notified
        return miPushMessage
    }

    @JvmStatic
    fun generateMessage(miPushMessage: MiPushMessage): PushMetaInfo {
        return PushMetaInfo().apply {
            setId(miPushMessage.messageId)
            setTopic(miPushMessage.topic)
            setDescription(miPushMessage.description)
            setTitle(miPushMessage.title)
            setNotifyId(miPushMessage.notifyId)
            setNotifyType(miPushMessage.notifyType)
            setPassThrough(miPushMessage.passThrough)
            setExtra(miPushMessage.extra)
        }
    }

    @JvmStatic
    fun getPushMode(context: Context): Int {
        if (pushMode == 0) {
            setPushMode(if (isUseCallbackPushMode(context)) PUSH_MODE_CALLBACK else PUSH_MODE_BROADCAST)
        }
        return pushMode
    }

    @JvmStatic
    fun isUseCallbackPushMode(context: Context): Boolean {
        val intent = Intent(PushConstants.MIPUSH_ACTION_NEW_MESSAGE)
        intent.setClassName(context.packageName, "com.xiaomi.mipush.sdk.PushServiceReceiver")
        return isIntentAvailable(context, intent)
    }

    @JvmStatic
    fun sendCommandMessageBroadcast(context: Context, miPushCommandMessage: MiPushCommandMessage) {
        val intent = Intent(PushConstants.MIPUSH_ACTION_NEW_MESSAGE).apply {
            setPackage(context.packageName)
            putExtra(MESSAGE_TYPE, MESSAGE_COMMAND)
            putExtra(KEY_COMMAND, miPushCommandMessage)
        }
        sdkGateway?.onReceivePushService(context, intent)
    }

    @JvmStatic
    fun sendQuitMessageBroadcast(context: Context) {
        val intent = Intent(PushConstants.MIPUSH_ACTION_NEW_MESSAGE).apply {
            setPackage(context.packageName)
            putExtra(MESSAGE_TYPE, MESSAGE_QUIT)
        }
        sdkGateway?.onReceivePushService(context, intent)
    }

    private fun isIntentAvailable(context: Context, intent: Intent): Boolean {
        return try {
            val receivers = context.packageManager.queryBroadcastReceivers(intent, 32)
            receivers.isNotEmpty()
        } catch (e: Exception) {
            true
        }
    }

    private fun setPushMode(mode: Int) {
        pushMode = mode
    }
}
